resources/scripts.js

```diff
@@ -2,88 +2,148 @@ function firstGame() {
         return;
     }
-    rollElement.disabled = true;
-}
-
-function betButtonState(enabled) {
-    maxBetInput.setAttribute('max', playerMoney);
-    displayMoney.innerHTML = playerMoney;
-    if (enabled === true) {
-        betDialog.disabled = false;
-        displayBetButton.disabled = false;
-        return;
+    betDialogElement.setAttribute('max', playerMoney);
+    lastPlayerMoney = playerMoney;
+    moneyChange(0);
+    rollButtonElement.disabled = false;
+    betDialogElement.disabled = true;
+    betButtonElement.disabled = true;
+    comeButtonElement.disabled = true;
+    if (pointOpen === true) {
+        comeButtonElement.disabled = false;
+        betDialogElement.disabled = false;
     }
-    betDialog.disabled = true;
-    displayBetButton.disabled = true;
+    return;
 }
 
-function firstRoll() {
+async function firstRoll() {
     diceRoll();
-    displayMessage(`You rolled ${playerRoll}`);
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    buildSummary(DICEROLLED);
     if (playerRoll === 7 || playerRoll === 11) {
-        playerWin();
+        payOut(true, true, betAmount);
         return;
     }
     if (playerRoll === 2 || playerRoll === 3 || playerRoll === 12) {
-        playerLose();
+        payOut(true, false);
         return;
     }
     playerPoint = playerRoll;
     pointOpen = true;
     buttonPosition(playerRoll);
-    displayMessage(`Your point is ${playerPoint}`);
+    buildSummary(SHOWPOINT);
+    displayMessage(rollSummary);
+    rollSummary = null;
+    buttonStates();
+    return;
 }
 
-function gameRoll() {
+async function gameRoll() {
     diceRoll();
-    displayMessage(`You rolled ${playerRoll}`);
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    buildSummary(DICEROLLED);
+    if (testObjPop(secondaryBetObj) === false) {
+        secondaryRoll(playerRoll);
+    }
     if (playerRoll === playerPoint) {
-        playerWin();
+        payOut(true, true, betAmount);
         return;
     }
     if (playerRoll === 7) {
-        playerLose();
+        payOut(true, false);
         return;
     }
-    displayMessage('No action, roll again');
+    buildSummary(NOACTION);
+    displayMessage(rollSummary);
+    rollSummary = null;
+    buttonStates();
+    return;
 }
-
 function checkGameState() {
     if (pointOpen === false) {
-       firstRoll(); 
-       return;
+        firstRoll();
+        return;
     }
     gameRoll();
+    return;
+}
+
+function secondaryRoll(roll) {
+    if (roll === 7) {
+        for (let i = 4; i < 11; i++) {
+            if (secondaryBetObj[i] > 0) {
+                buildSummary(SECONDARYLOSE, i);
+                payOut(false, false);
+                delete secondaryBetObj[i];
+            }
+        }
+    } else {
+        if (roll === 4 || roll === 5 || roll === 6 || roll === 8 || roll === 9 || roll === 10) {
+            buildSummary(PLACEWIN, roll);
+            payOut(false, true, secondaryBetObj[roll]);
```

Comment:
NaN issue appears to be on this line. Logic issue when come bet is set but no place. Will look into alternate way of dealing with this.

Reply:
NaN issue fixed.